client.go

```diff
@@ -61,6 +61,24 @@ func opts(opts ...RequestOption) *requestOptions {
 	return o
 }
 
+type ClientOption func(*Client)
+
+func WithHTTPClient(httpClient *http.Client) ClientOption {
+	return func(c *Client) {
+		c.httpClient = httpClient
+	}
+}
+
+func NewClientWithOptions(APIKey string, options ...ClientOption) *Client {
```

Comment:
You don't need to implement a new function `NewClientWithOptions` you can add the options to `NewClient`. As they are optional arguments existing code that call `NewClient` with the only with the API key will compile correctly even if you add `options ...ClientOption` to this function.

Reply:
You are right, if I was just calling `NewClient()` then I have no problems with code comilation, but consider next example:
```
func call(s string, f func(s string)) error { ... }

if err := call(token, vt.NewClient); err != nil { ... }
```
Because of signature of fuction was changed this code will never compile

Reply:
That case would certainly break. But I think the adoption of vt-go is still low and a case like this is be extremely rare, there's a very little chance that we break someone's code because of this. I prefer putting API simplicity over backward compatibility at this stage of the project.

Reply:
Agreed, thanks for reply. Closing PR.